microhapulator/qcsummary.py

```diff
@@ -0,0 +1,84 @@
+# -------------------------------------------------------------------------------------------------
+# Copyright (c) 2024, DHS.
+#
+# This file is part of MicroHapulator (https://github.com/bioforensics/microhapulator) and is
+# licensed under the BSD license: see LICENSE.txt.
+#
+# This software was prepared for the Department of Homeland Security (DHS) by the Battelle National
+# Biodefense Institute, LLC (BNBI) as part of contract HSHQDC-15-C-00064 to manage and operate the
+# National Biodefense Analysis and Countermeasures Center (NBACC), a Federally Funded Research and
+# Development Center.
+# -------------------------------------------------------------------------------------------------
+
+from .qcstats import FilterStats, PairedAmbiguityFilterStats, ReadMergingStats
+from dataclasses import dataclass
+
+
+@dataclass
+class SingleEndReadQCSummary:
+    ambig: FilterStats
+    length: FilterStats
```

Comment:
The table of QC stats for single-end data can be managed by a couple of simple `FilterStats` data structures, one for ambiguity filtering and another for length filtering. A few methods are defined on the class to make sure the right denominator is used for calculating the percentage of the total reads being filtered at each step.